src/main/java/taxi/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package taxi.service.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import taxi.dao.CarDao;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Car;
+import taxi.model.Driver;
+import taxi.service.CarService;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        carDao.create(car);
+        return car;
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).orElseThrow(()
+                -> new RuntimeException("No such car stored with id : " + id));
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        Car targetCar = carDao.get(car.getId()).orElseThrow(()
+                -> new RuntimeException("No such car stored : " + car.getModel()));
+        targetCar.getDrivers().add(driver);
+        carDao.update(targetCar);
+    }
```

Comment:
My idea was firstly to check if we have such car and then only to add if it is present, otherwise add a new driver. Of course for the sake of validation - we can do check for the `driver` as well.